syntax = "proto3"; //说明使用proto3语法定义协议

//package userOp;
option go_package = ".;userOp";     // go 中使用这种格式，指定包名

// 用户登录
message LoginRequest{
    int32 Id = 1;
    string userName = 2;
    string passwd = 3;
    bool rmbusrChk = 4;       // 是否记住用户名
    bool rmbpasswdChk = 5;    // 是否记住密码
}

// 登录返回信息
message LoginReply{
  int32 userId = 1;
  string LoginMgs = 2;    // 登录提示信息
  bool LoginSts = 3;      // 登录状态，True 表示成功，False 表示登录失败
}

// 用户注册
message UserRegisterRequest{
    int32 Id = 1;
    string userName = 2;
    string passwd = 3;
    string Eamil = 4;
}

message UserRegisterReply{
    int32 userId = 1;
    string RegisterMesg = 2;   // 注册提示消息
    bool RegisterSts = 3;      // 注册状态 True：成功， False：失败
}

service userOp{
    // 登录
    rpc UserLogin(LoginRequest) returns (LoginReply) {};
    // 注册
    rpc UserRegister(UserRegisterRequest) returns (UserRegisterReply) {};
}
